ts/state/ducks/groups.ts

```diff
@@ -0,0 +1,408 @@
+    });
+    builder.addCase(initNewGroupInWrapper.pending, (state, _action) => {
+      state.creationFromUIPending = true;
+
+      window.log.error('a initNewGroupInWrapper is pending');
+    });
+    builder.addCase(loadMetaDumpsFromDB.fulfilled, (state, action) => {
+      const loaded = action.payload;
+      loaded.forEach(element => {
+        state.infos[element.groupPk] = element.infos;
+        state.members[element.groupPk] = element.members;
+      });
+    });
+    builder.addCase(loadMetaDumpsFromDB.rejected, () => {
+      window.log.error('a loadMetaDumpsFromDB was rejected');
+    });
+    builder.addCase(refreshGroupDetailsFromWrapper.fulfilled, (state, action) => {
+      const { infos, members, groupPk } = action.payload;
+      if (infos && members) {
+        state.infos[groupPk] = infos;
+        state.members[groupPk] = members;
+
+        window.log.debug(`groupInfo after merge: ${stringify(infos)}`);
+        window.log.debug(`groupMembers after merge: ${stringify(members)}`);
+      } else {
+        window.log.debug(
+          `refreshGroupDetailsFromWrapper no details found, removing from slice: ${groupPk}}`
+        );
+
+        delete state.infos[groupPk];
+        delete state.members[groupPk];
+      }
+    });
+    builder.addCase(refreshGroupDetailsFromWrapper.rejected, () => {
+      window.log.error('a refreshGroupDetailsFromWrapper was rejected');
+    });
+    builder.addCase(destroyGroupDetails.fulfilled, (state, action) => {
+      const { groupPk } = action.payload;
+      // FIXME destroyGroupDetails marks the info as destroyed, but does not really remove the wrapper currently
+      delete state.infos[groupPk];
+      delete state.members[groupPk];
+    });
+    builder.addCase(destroyGroupDetails.rejected, () => {
+      window.log.error('a destroyGroupDetails was rejected');
+    });
+    builder.addCase(handleUserGroupUpdate.fulfilled, (state, action) => {
+      const { infos, members, groupPk } = action.payload;
+      if (infos && members) {
+        state.infos[groupPk] = infos;
+        state.members[groupPk] = members;
+
+        window.log.debug(`groupInfo after handleUserGroupUpdate: ${stringify(infos)}`);
+        window.log.debug(`groupMembers after handleUserGroupUpdate: ${stringify(members)}`);
+      } else {
+        window.log.debug(
+          `handleUserGroupUpdate no details found, removing from slice: ${groupPk}}`
+        );
+
+        delete state.infos[groupPk];
+        delete state.members[groupPk];
+      }
+    });
+    builder.addCase(handleUserGroupUpdate.rejected, () => {
+      window.log.error('a handleUserGroupUpdate was rejected');
+    });
+  },
+});
+
+export const groupInfoActions = {
+  initNewGroupInWrapper,
+  loadMetaDumpsFromDB,
+  destroyGroupDetails,
+  refreshGroupDetailsFromWrapper,
+  handleUserGroupUpdate,
+  ...groupSlice.actions,
+};
+export const groupReducer = groupSlice.reducer;
```

Comment:
This is very nice and clean. Hope we can get as much as possible to work like this in the future ™️

Reply:
yep, will try to do it this way going foward